src/shell.cpp

```diff
@@ -0,0 +1,107 @@
+#include "shell.h"
+
+Shell::Shell() {}
+
+void Shell::run() {
+    std::string input;
+
+    while (true) {
+        std::cout << "Shell> ";
+        std::getline(std::cin, input);
+
+        if (input == "exit") {
+            break;
+        }
+
+        if (input == "myjobs") {
+            updateBackgroundProcesses();
+            showBackgroundProcesses();
+            continue;
+        }
+
+        // Check if the command should be run in the background
+        bool runInBackground = false;
+        if (!input.empty() && input.back() == '&') {
+            runInBackground = true;
+            input.pop_back();
+        }
+
+        executeCommand(input, runInBackground);
+    }
+}
+
+void Shell::executeCommand(const std::string& command, bool runInBackground) {
+    pid_t pid = fork();
+    if (pid == -1) {
+        std::cerr << "Failed to create child process" << std::endl;
+        return;
+    } else if (pid == 0) {
+        // Child process
+        std::vector<std::string> args = splitString(command, ' ');
+
+        // Convert vector of strings to char* array
+        std::vector<char*> cArgs;
+        for (const auto& arg : args) {
+            cArgs.push_back(const_cast<char*>(arg.c_str()));
+        }
+        cArgs.push_back(nullptr); // Add a null terminator at the end
+
+        execvp(cArgs[0], cArgs.data());
+
+        // execvp will only return if an error occurred
+        std::cerr << "Failed to execute command: " << command << std::endl;
+        exit(EXIT_FAILURE);
+    } else {
+        // Parent process
+        if (runInBackground) {
+            BackgroundProcess process;
+            process.pid = pid;
```

Comment:
Why not implement a constructor for the struct that will allow you to set these attributes on creation?